src/commands/tag/set-missing.command.ts

```diff
@@ -0,0 +1,78 @@
+import { clone, Command, Input } from "../../deps.ts";
+import { CmdGlobalOptions } from "../cmd-options.ts";
+import { setupLogger } from "../../logger.ts";
+import { verifyCliAvailability } from "../../init.ts";
+import { loadConfig } from "../../config/config.model.ts";
+import { MeshAdapterFactory } from "../../mesh/mesh-adapter.factory.ts";
+import { MeshAdapter } from "../../mesh/mesh-adapter.ts";
+import { MeshTenant } from "../../mesh/mesh-tenant.model.ts";
+import { MeshInvalidTagValueError } from "../../errors.ts";
+
+export const setMissingTags = new Command()
+  .description(
+    "Fix all tenants missing the given tag. Collie will ask you per tenant what the value should be and writes the change to the cloud platform(s).",
+  )
+  .action(setMissingTagsAction);
+
+async function setMissingTagsAction(options: CmdGlobalOptions, tagKey: string) {
+  setupLogger(options);
+  await verifyCliAvailability();
+
+  const config = loadConfig();
+  const meshAdapterFactory = new MeshAdapterFactory(config);
+  const meshAdapter = meshAdapterFactory.buildMeshAdapter(options);
+
+  const allTenants = await meshAdapter.getMeshTenants();
+
+  const tenantsMissingTag = allTenants.filter((x) =>
+    !x.tags.find((y) => y.tagName === tagKey)
+  );
+
+  console.log(
+    `We have identified ${tenantsMissingTag.length} tenants without the tag "${tagKey}". You'll be able to set the tag values for these tenants now.`,
+  );
+
+  let count = 1;
+  for (const tenant of tenantsMissingTag) {
+    await askAndSetTag(
+      tagKey,
+      tenant,
+      meshAdapter,
+      count,
+      tenantsMissingTag.length,
+    );
+    count++;
+  }
+}
+
+async function askAndSetTag(
+  tagKey: string,
+  tenant: MeshTenant,
+  meshAdapter: MeshAdapter,
+  count: number,
+  totalCount: number,
+) {
+  const prefix = `(${count}/${totalCount})`;
+  const tagValue: string = await Input.prompt({
+    message:
+      `${prefix} For the tenant ${tenant.platformTenantName} (${tenant.platformTenantId}) on ${tenant.platform}, what tag value do you want to set for "${tagKey}"? (Leave empty to skip)`,
+  });
+
+  if (tagValue !== "") {
+    // Skip this tenant if the user does not give any value.
+    const originalTenant = clone(tenant);
```

Comment:
d: oh this is dangerous and could easily be forgotten or done wrong later on. We must make sure that we return cloned tenants from the cache and the other meshAdapter layers, this should work. But using your pattern you still need an unmodified tenant for your API design to work. I suggest to set up a copyTenant function as the MeshTenant is an interface and can not contain methods. They way its at least clear that this function is meant to be used in conjunction with the interface. My biggest problem here is: its not always clear if this really is a deep copy or a shallow copy. (we need a deep copy).
Best would be to have a copy function on the tenant object itself but I don't really want to drop the interface nature of `MeshTenant` as this really makes loading and saving a lot easier. So the next best thing which comes to my mind is a dedicated copy function.

Reply:
okay this is doing a deep clone. Checked the dep itself. So I guess its a bit senseless to wrap it just "because" in another function.

Reply:
Yup I took care of this already. I think the `clone` keyword is good enough although we can rename it to something else if you want in the `deps.ts` file. I think `clone` is fine.